### PR TITLE
[FEATURE] Add list of available languages for code blocks

### PR DESCRIPTION
**Pull request by brotkrueml**

### CODE REVIEW

**Comment by brotkrueml** [comment]
This is a draft for outputting the available languages for a code-block directive.

This is meant for usage in
https://docs.typo3.org/m/typo3/docs-how-to-document/main/en-us/WritingReST/Reference/Code/Codeblocks.html#available-lexers

It should be used only internal in our docs.

@linawolf What do you think?

**Comment by linawolf** [comment]
@brotkrueml I like the idea!

**Comment by brotkrueml** [comment]
@linawolf Then now it can be reviewed :-)

File: tests/Integration/tests/codehighlight-languages/input/index.rst [hunk 1]
```diff
@@ -0,0 +1,7 @@
+===============================
+Registered code-block languages
+===============================
+
+All registered languages for code blocks:
+
+..  codeblock-languages::
```
Comment (linawolf) [inline tests/Integration/tests/codehighlight-languages/input/index.rst:7]: I would suggest that we use a common domain for directives that are used to document our rst-implementation. Maybe 

rest:codeblock-languages

Ist seems like we are gonna have more then just one

Comment (brotkrueml) [reply tests/Integration/tests/codehighlight-languages/input/index.rst:7]: This is valid?

    .. rest:codeblock-languages::

How do I name then the according template respective the path?

Comment (linawolf) [reply tests/Integration/tests/codehighlight-languages/input/index.rst:7]: just like we do in the php domain?

Comment (linawolf) [reply tests/Integration/tests/codehighlight-languages/input/index.rst:7]: directive/rest/codeblock-languages.html.twig

Comment (brotkrueml) [reply tests/Integration/tests/codehighlight-languages/input/index.rst:7]: Why "rest"? Maybe "t3" would be better as it is more an internal directive?

Comment (brotkrueml) [reply tests/Integration/tests/codehighlight-languages/input/index.rst:7]: Or "internal"? That's make the purpose clear.

Comment (linawolf) [reply tests/Integration/tests/codehighlight-languages/input/index.rst:7]: h2document?

Comment (garvinhicking) [reply tests/Integration/tests/codehighlight-languages/input/index.rst:7]: OMG, is "h2document" = "how to document"? This is so not clear :-P

I would vote for "guides", "t3", "internal" in that order...

Comment (linawolf) [reply tests/Integration/tests/codehighlight-languages/input/index.rst:7]: I like guides. I would reserve t3 for typo3 specific stuff, like if we ever want to do something expecially for TypoScript or whatever?

Comment (brotkrueml) [reply tests/Integration/tests/codehighlight-languages/input/index.rst:7]: Okay, will use guides :1st_place_medal:

Comment (brotkrueml) [reply tests/Integration/tests/codehighlight-languages/input/index.rst]: Done, please have a look.

**Review by garvinhicking — APPROVED** [review]
Very cool!

Next up, `guides:inventories`? 😇😶‍🌫️

**Review by linawolf — APPROVED** [review]

**Pull request merged**